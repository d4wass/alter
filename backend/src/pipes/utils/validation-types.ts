import { ValidationError } from '@nestjs/common';

export type ConvertedValidationError = {
  property: string;
  constraints: any;
  children?: ValidationError[];
};

export type ConstructorTypes =
  | StringConstructor
  | BooleanConstructor
  | NumberConstructor
  | ArrayConstructor
  | ObjectConstructor;
