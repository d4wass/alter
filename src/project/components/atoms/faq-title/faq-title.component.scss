.title-wrapper {
    display: flex;
    align-items: center;
    h2{
        margin-right: 10px;
    }
}