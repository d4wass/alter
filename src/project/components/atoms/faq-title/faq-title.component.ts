import { Component, Input } from '@angular/core';

@Component({
  selector: 'app-faq-title',
  template: `
    <div class="title-wrapper">
      <h2>0{{ titleNumber + 1 }}</h2>
      <h3>{{ title }}</h3>
    </div>
  `,
  styleUrls: ['./faq-title.component.scss']
})
export class FaqTitleComponent {
  @Input() title!: string;
  @Input() titleNumber!: number;
}
