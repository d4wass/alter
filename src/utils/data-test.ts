export function dataTest(selector: string) {
  return `[data-test="${selector}"]`;
}
