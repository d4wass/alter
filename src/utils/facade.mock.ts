import { Observable, BehaviorSubject } from 'rxjs';

export type FacadeMock<T> = {
  [P in keyof T]: T[P] extends Observable<infer U>
    ? BehaviorSubject<U>
    : T[P] extends (args: infer A) => Observable<infer R>
    ? (args: A) => BehaviorSubject<R>
    : T[P];
};
