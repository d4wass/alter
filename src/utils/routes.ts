export interface urlItem {
  description: string;
  url: string;
}

export const urlList: urlItem[] = [
  { description: 'Learn more', url: 'faq' },
  { description: 'Carculator', url: 'carculator' },
  { description: 'Become a host', url: 'host' },
  { description: 'Create Account', url: 'new-user' },
  { description: 'Your Cars', url: 'vehicles' }
];
