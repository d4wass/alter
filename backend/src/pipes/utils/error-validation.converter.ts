import { ValidationError } from 'class-validator';
import { ConvertedValidationError } from './validation-types';

export class ValidationErrorConverter {
  private _convertedErrors: Array<ConvertedValidationError> = [];
  public get convertedErrors(): Array<ConvertedValidationError> {
    return this._convertedErrors;
  }
  public set convertedErrors(value: Array<ConvertedValidationError>) {
    this._convertedErrors = value;
  }
  private _errors: ValidationError[];

  constructor(errors: ValidationError[]) {
    this._errors = errors;
    this.errorsConverter(this._errors);
  }

  errorsConverter(errors: ValidationError[]): void {
    errors.forEach((error) => {
      if (!error.children.length) {
        this._convertedErrors.push({ property: error.property, constraints: error.constraints });
      } else {
        this.errorsConverter(error.children);
      }
    });
  }
}
