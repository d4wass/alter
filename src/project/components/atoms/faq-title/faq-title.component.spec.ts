import { createComponentFactory, Spectator } from '@ngneat/spectator';
import { dataTest } from '../../../../utils/data-test';
import { FaqTitleComponent } from './faq-title.component';

describe('FaqTitleComponent', () => {
  let spectator: Spectator<FaqTitleComponent>;
  let component: FaqTitleComponent;

  const createComponent = createComponentFactory(FaqTitleComponent);
  const selectors = {
    title: () => spectator.query(dataTest('titleNumber'))?.textContent
  };

  beforeEach(() => {
    spectator = createComponent();
    component = spectator.component;
  });

  it('should match to snapshot', () => {
    expect(component).toMatchSnapshot();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should show changed value in <h2> tag', () => {
    component.titleNumber = 11;
    spectator.detectComponentChanges();

    expect(selectors.title()).toEqual('012');
  });

  describe('@Inputs', () => {
    it('should set [title] to passed value', () => {
      component.title = 'title';
      spectator.detectComponentChanges();

      expect(component.title).toEqual('title');
    });
    it('should set [titleNumber] to passed value', () => {
      component.titleNumber = 20;
      spectator.detectComponentChanges();

      expect(component.titleNumber).toEqual(20);
    });
  });
});
