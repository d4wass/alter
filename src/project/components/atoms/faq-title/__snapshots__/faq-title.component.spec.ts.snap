// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FaqTitleComponent should match to snapshot 1`] = `
FaqTitleComponent {
  "__ngContext__": 0,
}
`;
